packages/frontend/src/components/Explorer/CustomMetricModal/index.tsx

```diff
@@ -0,0 +1,251 @@
+            ? item.filters?.map((filterRule) =>
+                  addFieldIdToMetricFilterRule(filterRule),
+              ) || []
+            : [];
+    }, [isEditing, item]);
+
+    const [customMetricFiltersWithIds, setCustomMetricFiltersWithIds] =
+        useState<MetricFilterRuleWithFieldId[]>(
+            initialCustomMetricFiltersWithIds,
+        );
+
+    useEffect(() => {
+        setCustomMetricFiltersWithIds(initialCustomMetricFiltersWithIds);
+    }, [initialCustomMetricFiltersWithIds]);
+
+    const handleOnSubmit = form.onSubmit(({ customMetricLabel }) => {
+        if (!item) return;
+
+        const data = prepareCustomMetricData({
+            dimension: item,
+            type: customMetricType!,
+            customMetricLabel,
+            customMetricFiltersWithIds,
+            isEditingCustomMetric: !!isEditing,
+            item,
+            exploreData,
+        });
+
+        if (isEditing && isAdditionalMetric(item)) {
+            editAdditionalMetric(
+                {
+                    ...item,
+                    ...data,
+                },
+                getFieldId(item),
+            );
+            showToastSuccess({
+                title: 'Custom metric edited successfully',
+            });
+        } else if (isDimension(item) && form.values.customMetricLabel) {
+            addAdditionalMetric({
+                uuid: uuidv4(),
+                table: item.table,
+                sql: item.sql,
+                type: customMetricType!,
+                baseDimensionName: item.name,
+                ...data,
+            });
+            showToastSuccess({
+                title: 'Custom metric added successfully',
+            });
+        }
+        toggleModal();
+    });
+
+    const defaultFilterRuleFieldId = useMemo(() => {
+        if (item) {
+            if (!isEditing) return getFieldId(item);
+
+            if (
+                isEditing &&
+                'baseDimensionName' in item &&
+                item.baseDimensionName
+            ) {
+                return `${item.table}_${item.baseDimensionName}`;
+            }
+        }
+    }, [isEditing, item]);
+
+    return item ? (
+        <Modal
+            size="xl"
+            centered
+            zIndex={15}
+            onClick={(e) => e.stopPropagation()}
+            opened={isOpen}
+            onClose={() => toggleModal(undefined)}
+            title={
+                <Title order={4}>
+                    {isEditing ? 'Edit' : 'Create'} Custom Metric
+                </Title>
+            }
+        >
+            <form onSubmit={handleOnSubmit}>
+                <Stack>
+                    <TextInput
+                        label="Label"
+                        required
+                        placeholder="Enter custom metric label"
+                        {...form.getInputProps('customMetricLabel')}
+                    />
+                    {defaultFilterRuleFieldId && (
```

Comment:
We should allow filters to be added to old custom metrics. Or is there any reason why they shouldn't work ?

Reply:
🤦 Exactly, they should work regardless! 
Reverting this change.